data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/stream/ShardConsumer.java

```diff
@@ -152,6 +157,11 @@ public Builder tableInfo(TableInfo tableInfo) {
             return this;
         }
 
+        public Builder shardId(final String shardId) {
```

Comment:
Since you add this, maybe it's worth adding a info log every time there is a regular checkpointing?  As still, there won't be any logs in couple of hours processing until something went wrong or completed. Not sure if the feedback is that people like to know it's still live..

Reply:
I considered this yeah. I guess a log every 2 minutes isn't too much. Can we log anything else in this message? Records written to buffer?